### ISSUE TITLE
Resume in the Segmentation example

### ISSUE THREAD

**Issue by netw0rkf10w** [pytorch/vision#2116]
## 🐛 Bug

The segmentation training [script](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py) doesn't seem to allow correct resuming. I think [lines 131-133](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L131) should be changed to:

```
    start_epoch = 0
    if args.resume:
        checkpoint = torch.load(args.resume, map_location='cpu')
        model.load_state_dict(checkpoint['model'])
        optimizer.load_state_dict(checkpoint['optimizer'])
        lr_scheduler.load_state_dict(checkpoint['lr_scheduler'])
        start_epoch = checkpoint['epoch'] + 1
```

Then, [line 161](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L161) and [lines 167-173](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L167) should be changed respectively to:

```
    for epoch in range(start_epoch, args.epochs):
```

and

```
utils.save_on_master(
            {
                'model': model_without_ddp.state_dict(),
                'optimizer': optimizer.state_dict(),
                'lr_scheduler': lr_scheduler.state_dict(),
                'epoch': epoch
            },
```

Besides, is there a good reason to set [`batch_size=1`](https://github.com/pytorch/vision/blob/7b60f4db9707d7afdbb87fd4e8ef6906ca014720/references/segmentation/train.py#L120) (instead of `args.batch_size` or even `2*args.batch_size`) for `data_loader_test`?

Thanks.

**Comment by netw0rkf10w** [comment]
> Besides, is there a good reason to set `batch_size=1` for `data_loader_test`?

As I understand this is a hack to prevent the test data loader from having duplicates as `torch.utils.data.distributed.DistributedSampler` always fills the dataset so that the total number of samples is divisible by `num_gpus*batch_size`. However, if the number of validation samples is not a multiple of `num_gpus` then the results will not be correct even if `batch_size=1`.

**Comment by vincentqb** [comment]
The point you mention about resuming looks good to me. Do you want to open a pull request to address these? Are there other training script you noticed that had that?

Removing 'args' from the checkpoint is BC breaking though. We could decide not to load it when loading the checkpoint.

**Comment by netw0rkf10w** [comment]
@vincentqb Thanks. I have checked the other examples and they don't have the issue. I created a PR here: https://github.com/pytorch/vision/pull/2141. The code is more consistent with the other examples than what I proposed above.

**Comment by netw0rkf10w** [comment]
@vincentqb @fmassa Regarding the evaluation data loader issue, it seems that there's no easy fix. Currently **the only way to get the validation accuracy right is to use only 1 GPU**. For multiple GPUs, both `DataParallel` and `DistributedDataParallel` create duplicates. Maybe we should clarify this in the README. Note that this issue affects all the examples and not only Segmentation.

**Comment by fmassa** [comment]
@netw0rkf10w hi,

Great questions.

About the second one (evaluation of the data loader), indeed this was an approximation that I did for both the classification and the segmentation reference training scripts.
Note that the detection training script handles it correctly, as it keeps the evaluation results for each image in a dict https://github.com/pytorch/vision/blob/9329257beeb51f7855cc765df15b0be21e361133/references/detection/engine.py#L95
so that duplicate entries don't get counted twice. But this would make the reference scripts for classification and segmentation more complicated.

A PR improving the README of the `classification` and `segmentation` examples would be great! Can you send a PR?
